migrator/src/main/java/com/google/phonenumbers/migrator/MigrationJob.java

```diff
@@ -0,0 +1,160 @@
+package com.google.phonenumbers.migrator;
+
+import com.google.i18n.phonenumbers.metadata.RangeSpecification;
+import com.google.i18n.phonenumbers.metadata.RangeTree;
+import com.google.i18n.phonenumbers.metadata.i18n.PhoneRegion;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Represents a migration operation for a given region where each {@link MigrationJob} contains
+ * a {@link RangeTree} of numbers to be migrated as well as the {@link CsvTable} which will hold the
+ * available recipes that can be performed on the range. Only recipes from the given regionCode will
+ * be used.
+ */
+public final class MigrationJob {
```

Comment:
I like the idea of having this class could be separated into two; one being a MigratorFactory which creates and returns a Migrator object and then another class defining the Migrator object.

The MigrationFactory would just be responsible to creating the migrator objects, so its only public facing methods would be:

`public Migrator createMigrator(String numberInput, String regionCode) {}` and others for whichever inputs you have to create a Migrator.

I find that factories like this make it a bit easier to understand the responsibilities of the classes and we use them quite a bit on our team internally.

Also, the getMigratableNumbers methods could live in the Migrator class. Or later on, you could maybe create a utils class if you find you have a lot of utility methods.